### DISCUSSION TITLE
dockerignore behaving differently in podman vs docker

### DISCUSSION THREAD

**Discussion by thomasmattheussen** [containers/podman#22910] [Q&A]
I have the following dockerignore which produces different results when building with podman vs building with docker.

```
*
!packages/my-theme-*/package.json
```

and a directory listing such as

packages/my-theme-a/package.json
packages/my-theme-b/package.json
packages/my-theme-c/package.json

You can easily test by running this command (once with docker and once with podman)

```
docker build --no-cache --progress plain --file - . <<EOF
FROM busybox
COPY . /build-context
WORKDIR /build-context
RUN find .
EOF
```

When using docker, it does what I would expect (exclude everything except those files). When using podman, it just excludes everything. It seems as though the files I mentioned in dockerignore are somehow not matched.

I have tried several patterns such as these below but none seem to actually work.

```
!packages/**/package.json
```

```
!**/package.json
```

(I'm using the latest podman 5.1.1)

**Comment by nalind** [comment]
Huh.  It looks like .dockerignore behavior change between the "classic" builder and the BuildKit-based one - I get the same results if I run the docker build with DOCKER_BUILDKIT=0 set in my environment.  We're going to want to match the newer behavior sooner or later, though.

**Reply by rhatdan** [reply]
Yes I think we should default ot matching buildkit

**Reply by thomasmattheussen** [reply]
Thanks for your reply! Not sure how to fix my problem though 😅 
I'd like for my `podman build` to get the same results as with the `docker build`. It seems the wildcard matching just isn't working correctly right now, or am I misunderstanding something?

**Reply by afbjorklund** [reply]
You can use buildkit with podman, but unfortunately not with `podman buildx`

https://github.com/moby/buildkit#podman

**Reply by stevekuznetsov** [reply]
@nalind @rhatdan any chance y'all have any bandwidth to handle this? `podman build` is unusable for any newer project that expects BuildKit behavior, which seems to be default now for `docker`.

**Comment by landsman** [comment]
It just does not work, do not respect `.dockerignore` at all 😢 

- using docker compose (`/Users/landsman/bin/docker-compose`).
- I got the same results with `podman-compose`

```
podman-compose version 1.3.0
podman version 5.3.2
```

**Reply by mikarwacki** [reply]
Same for me the wildcard skips every file and seems to override any further statements in dockerignore

**Comment by avikivity** [comment]
Also seeing my .dockerignore being ignored. strace shows it's being read.
